### PR TITLE
fix: update Uname parser to fix LooseVersion comparision error

### PR DESCRIPTION
**Pull request by qinpingli**
Detailed error is:
TypeError: '<' not supported between instances of 'str' and 'int'

### All Pull Requests:

Check all that apply:

* [X] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [x] Is this PR to correct an issue?
* [X] Is this PR an enhancement?

### Complete Description of Additions/Changes:

The issue is: https://github.com/RedHatInsights/insights-core/issues/3815

### CODE REVIEW

File: insights/parsers/uname.py [hunk 1]
```diff
@@ -397,6 +397,24 @@ def _best_version(self, other):
...
+            pad_release(".".join(s_release_parts[:-1]), len(s_release_parts))
+            if s_release_parts[-1].startswith("el")
+            else ".".join(s_release_parts)
+        )
+        o_release = (
+            pad_release(".".join(o_release_parts[:-1]), len(o_release_parts))
+            if o_release_parts[-1].startswith("el")
+            else ".".join(o_release_parts)
+        )
+        return LooseVersion(s_release), LooseVersion(o_release)
```
Comment (xiangce) [inline insights/parsers/uname.py:416]: Better `show` (not `raise`) a warning when only one of the operands contains the `distribution`. And when both of them contain `distribution`, we can take them into comparison. E.g.

```suggestion
        dist_opts = ('el',)
        s_release_parts = self._lv_release.vstring.split(".")
        o_release_parts = other._lv_release.vstring.split(".")
        s_dist = s_release_parts[-1]
        o_dist = o_release_parts[-1]
        s_release = ".".join(s_release_parts)
        o_release = ".".join(o_release_parts)
        if s_dist.startswith(dist_opts) ^ o_dist.startswith(dist_opts):
            import warnings
            warnings.warn('Comparison of distribution part will be ingored.')
            s_release = (
                s_release if not s_dist.startswith(dist_opts)
                else pad_release(".".join(s_release_parts[:-1]), len(s_release_parts))
            )
            o_release = (
                o_release if not o_dist.startswith(dist_opts)
                else pad_release(".".join(o_release_parts[:-1]), len(o_release_parts))
            )
        return LooseVersion(s_release), LooseVersion(o_release)
```

File: insights/parsers/uname.py [hunk 2]
```diff
@@ -603,6 +630,12 @@ def pad_release(release_to_pad, num_sections=4):
     If the number of sections of the release to be padded is
     greater than num_sections, a ``ValueError`` will be raised.
     '''
+    def is_rt_release(items):
```
Comment (xiangce) [inline insights/parsers/uname.py:633]: Generally, a function name starting with `is_xxx` always return a `Boolean` value, but here not. Please update the name properly.

**Review by xiangce — CHANGES_REQUESTED** [review]
Since the `pad_release` method is also updated, please add tests to cover the new updates.

File: insights/parsers/uname.py (hunk 2 above)
Comment (qinpingli) [reply insights/parsers/uname.py]: Thanks, updated.

File: insights/parsers/uname.py (hunk 1 above)
Comment (qinpingli) [reply insights/parsers/uname.py]: Thanks, updated, please help take a look again.

**Comment by xiangce** [comment]
> Since the `pad_release` method is also updated, please add tests to cover the new updates.

@qinpingli - you may miss this comment? and btw, since a `warning` message was added to show for the `fixed_by`, better add more tests for it too.

**Comment by qinpingli** [comment]
> > Since the `pad_release` method is also updated, please add tests to cover the new updates.
> 
> @qinpingli - you may miss this comment? and btw, since a `warning` message was added to show for the `fixed_by`, better add more tests for it too.

You are right! Sorry for missing it. Now, new test cases are added, please help take a look again, thanks!

File: insights/tests/parsers/test_uname.py [hunk 3]
```diff
@@ -236,19 +239,42 @@ def test_fixed_by():
     assert ['2.6.33-100.el6'] == u.fixed_by('2.6.33-100.el6')
     assert ['2.6.32-600.el6'] == u.fixed_by('2.6.32-220.1.el6', '2.6.32-600.el6')
 
+    # For all remaining tests, cause the warnings to always be caught
```
Comment (xiangce) [inline insights/tests/parsers/test_uname.py:242]: Better add a new test function to wrap the following tests for the warnings.

And better call `warnings.resetwarnings()` to reset the warning filters after the test. (at the end of a new added tet function.

Comment (qinpingli) [reply insights/tests/parsers/test_uname.py:243]: Thanks, updated.

**Review by xiangce — APPROVED** [review]

File: insights/tests/parsers/test_uname.py (hunk 3 above)
Comment (qinpingli) [reply insights/tests/parsers/test_uname.py]: @xiangce thanks, updated.

**Pull request merged**